examples/step-1/doc/results.dox

```diff
@@ -34,6 +34,80 @@ good for some things at least.)
 
+<h4> Use triangles </h4>
+
+For the first 20 or so years of its existence, deal.II only supported
+hypercube elements (i.e., quadrilaterals in 2d, and hexahedra in
+3d). It now also supports triangles in 2d; and tetrahedra, pyramids,
+and wedges in 3d. A consequence of this history is that nearly all of
+the tutorial programs you will see exclusively use quadrilaterals and
+hexahedra, and you may be forgiven that that is all that's
+supported. But you can try out other types of cells yourself here
+already. For example, here are two ideas:
+
+- You could create a triangular triangulation meshing a triangular
+  domain. To do this, you would replace the call to
+  `GridGenerator::hyper_cube(triangulation);` in the `first_grid()`
+  function by `GridGenerator::reference_cell(triangulation,
+  ReferenceCells::Triangle);`. This will give you the following output
+  in `grid-1.svg`:
+  <img src="https://www.dealii.org/images/steps/developer/step-1.grid-1-triangle.png" alt="">
+
+- You can start with a quadrilateral mesh and convert it to a
+  triangular mesh. For example, in the `first_grid()` function,
+  replace the code
+  @code
+  Triangulation<2> triangulation;
+  GridGenerator::hyper_cube(triangulation);
+  @endcode
+  by the following, which first creates a temporary mesh
+  `triangulation_quad` consisting of quadrilaterals, and then converts
+  it into the `triangulation` object that then only consists of
+  triangles:
+  @code
+  Triangulation<2> triangulation_quad;
+  GridGenerator::hyper_cube(triangulation_quad);
+  Triangulation<2> triangulation;
+  GridGenerator::convert_hypercube_to_simplex_mesh (triangulation_quad,
+                                                    triangulation);
+  @endcode
+  This produces the following mesh:
+  <img src="https://www.dealii.org/images/steps/developer/step-1.grid-1-triangle-2.png" alt="">
+
+  You can do the same in the `second_grid()` function by replacing
+  @code
+  Triangulation<2> triangulation;
+
+  const Point<2> center(1, 0);
+  const double   inner_radius = 0.5, outer_radius = 1.0;
+  GridGenerator::hyper_shell(
+    triangulation, center, inner_radius, outer_radius, 10);
+  @endcode
+  by the following (that includes some magic at the bottom we're
+  perhaps not quite ready to explain in detail yet, but that is
+  mentioned in the documentation of
+  GridGenerator::convert_hypercube_to_simplex_mesh()):
+  @code
+  Triangulation<2> triangulation_quad;
+
+  const Point<2> center(1, 0);
+  const double   inner_radius = 0.5, outer_radius = 1.0;
+  GridGenerator::hyper_shell(
+    triangulation_quad, center, inner_radius, outer_radius, 10);
+
+  Triangulation<2> triangulation;
+  GridGenerator::convert_hypercube_to_simplex_mesh (triangulation_quad,
+                                                    triangulation);
+  for (const auto i : triangulation_quad.get_manifold_ids())
+    if (i != numbers::flat_manifold_id)
+      triangulation.set_manifold(i, triangulation_quad.get_manifold(i));
```

Comment:
This is a bit unfortunate to have in the first tutorial. Isn't it possible to attache a spherical manifold?

Reply:
Yes, that's why a say the thing about "magic we're not yet ready to explain" above. I think attaching a manifold would not be clearer either, though. We only get to manifolds in step-9.

Do you remember why the function that subdivides cells into triangles doesn't do this itself? The documentation says you have to do it, so why not do it internally?

Reply:
Ah, good point. Yes, not much we can do about it.

My inclination is to leave the text as is, including stating that we're doing some magic, because it doesn't require substantial changes for people to play around with.

Reply:
Would it make sense to introduce a helper function in GridTools that we call here? It would reduce the amount of magic to a single function call.